### PR TITLE
[AIRFLOW-4438] Add Gzip compression to S3_hook

### PR DESCRIPTION
**Pull request by OmairK**
- Added bool parameter gzip to load_file to s3_hook
 - Tested the load_file with load_file_gzip unittest
 - Updated the load_file docstring to reflect the extra parameter

---
Issue link: [AIRFLOW-4438](https://issues.apache.org/jira/browse/AIRFLOW-4438)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 1]
```diff
@@ -442,6 +445,9 @@ def load_file(self,
         :param encrypt: If True, the file will be encrypted on the server-side
             by S3 and will be stored in an encrypted form while at rest in S3.
         :type encrypt: bool
+        :param gzip: If True, the file will be compressed on the server-side
```
Comment (potiuk) [inline airflow/providers/amazon/aws/hooks/s3.py]:
```suggestion
        :param gzip: If True, the file will be compressed locally
```

**Review by potiuk — APPROVED** [review]
Just one nit-picking. If you correct it - we can merge it :)

File: airflow/providers/amazon/aws/hooks/s3.py (hunk 1 above)
Comment (OmairK) [reply airflow/providers/amazon/aws/hooks/s3.py]: Got it, I will resolve this

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks @OmairK !

**Comment by OmairK** [comment]
Your welcome @potiuk  :smile: . Thanks a lot for guiding me through my first PR.

**Comment by thierryturpin** [comment]
Hello, @OmairK 

I'm afraid this isn't working like it should. When you pass a string filename, how can filename.name be derived? https://github.com/apache/airflow/pull/7680/files#diff-411e5140735ea97206a32c7b390e2800R459

Or I'm misunderstanding how this should work.

thx for update.

**Comment by OmairK** [comment]
@thierryturpin Thanks for pointing it out, you are right about this I overlooked this silly mistake. I have made the changes and PR has been merged #8571

**Pull request merged**